File: Back/src/main/java/com/BackPM/BackPM/models/Pedido.java
package com.BackPM.BackPM.models;

import jakarta.persistence.*;
import java.util.Date;
import java.util.List;
import java.util.ArrayList;

@Entity
@Table(name = "pedido")
public class Pedido extends ABaseEntity {

    @Column(name = "fecha")
    private Date fecha;

    @Column(name = "estado")
    private Boolean estado;

    @ManyToMany
    @JoinTable(
        name = "pedido_producto",
        joinColumns = @JoinColumn(name = "pedido_id"),
        inverseJoinColumns = @JoinColumn(name = "producto_id")
    )
    private List<Producto> productos = new ArrayList<>();

    // Getters y setters

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public Boolean getEstado() {
        return estado;
    }

    public void setEstado(Boolean estado) {
        this.estado = estado;
    }

    public List<Producto> getProductos() {
        return productos;
    }

    public void setProductos(List<Producto> productos) {
        this.productos = productos;
    }
}

File: frontend/src/models/Pedido.ts
import { Producto } from "./Producto";

export class Pedido {
  id: number;
  fecha: string;
  estado: boolean;
  productoId: Producto;

  constructor(id: number, fecha: string, estado: boolean, productoId: Producto) {
    this.id = id;
    this.fecha = fecha;
    this.estado = estado;
    this.productoId = productoId;
  }
}
